src/translationsmanager.cpp

```diff
@@ -0,0 +1,58 @@
+#include "translationsmanager.h"
+#include <QGuiApplication>
+
+TranslationsManager::TranslationsManager(QObject *parent)
+    : QObject{parent}, m_currentTranslator{nullptr} {
+  m_enTranslator.load(QLocale{QLocale::English}, QLatin1String("mysthea"),
+                      QLatin1String("_"), QLatin1String(":/translations"));
+
+  m_itTranslator.load(QLocale{QLocale::Italian}, QLatin1String("mysthea"),
+                      QLatin1String("_"), QLatin1String(":/translations"));
+
+  m_frTranslator.load(QLocale{QLocale::French}, QLatin1String("mysthea"),
+                      QLatin1String("_"), QLatin1String(":/translations"));
+
+  m_deTranslator.load(QLocale{QLocale::German}, QLatin1String("mysthea"),
+                      QLatin1String("_"), QLatin1String(":/translations"));
+
+  setCurrentLanguage(SupportedLanguage::English);
+}
+
+void TranslationsManager::setCurrentLanguage(SupportedLanguage language) {
+  switch (language) {
```

Comment:
There must not be need to remove and install translators if the language being set is equals to current one.